packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,243 @@
+ * however, it is possible to specify a different value using the customSize property.
+ */
+export const avatarSizeValues = [20, 24, 28, 32, 36, 40, 48, 56, 64, 72, 96, 120, 128] as const;
+export type AvatarSizeValue = typeof avatarSizeValues[number]; // 20 | 24 | 28 | ... | 128
+// !! Important: when adding new AvatarSizeValues, add corresponding "._size" classes in Avatar.scss
+
+/** Default Avatar size if not specified */
+export const defaultAvatarSize: AvatarSizeValue = 32;
+
+}
+```
+
+### Styling Tokens
+
+```tsx
+export type AvatarTokenSet = {
+  /** Width of the avatar */
+  width?: string;
+
+  /** Height of the avatar */
+  height?: string;
+
+  /** Background shown behind the initials or icon */
+  background?: string;
+
+  /** Color of the initials or icon */
+  color?: string;
+
+  /** Border radius */
+  borderRadius?: string;
+
+  /** Font used by the initials */
+  fontFamily?: string;
+
+  /** Font size used by the initials */
+  fontSize?: string;
+
+  /** Font weight used by the initials */
+  fontWeight?: string;
+
+  /** Font size used by the icon */
+  iconSize?: string;
+
+  /** Color of the ring when active=true and activeDisplay includes 'ring' */
+  activeRingColor?: string;
+
+  /** Color of the glow when active=true and activeDisplay includes 'glow' */
+  activeGlowColor?: string;
+
+  /** Opacity of the avatar when active=false */
+  inactiveOpacity?: string;
+
+  /** Scale transform applied to the avatar when active=false */
+  inactiveScale?: string;
+};
+```
+
+## Structure
+
+```html
+<span class="root">
+  <!-- Initials or icon -->
+  <span class="label">AB</span>
+  <!-- Profile image, if given -->
+  <img class="image" src="..." />
+  <span class="badge">
+    <!-- Content of the badge slot goes here, if given -->
+  </span>
+</span>
+```
+
+## Migration
+
+See [MIGRATION.md](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/MIGRATION.md).
+
+## Behaviors
+
+### States
+
+- **Display** - The Avatar will respect the `display` property to pick what content to show, if specified. Otherwise, it'll use the following priority:
+
+  - The `image` property.
+  - If no image is available (or it can't be loaded), then it will display initials derived from the `name` property.
+  - If no initials are available, display the provided `icon`.
+  - If no icon is provided, the default "person" icon will be used.
+
+- **Active** - The `active` property affects the display of the avatar if set. There will be an animation when switching between active and inactive.
+  - `(unset)` - Display at normal size/opacity.
+  - `false` - Reduce to 80% opacity, and 87.5% size.
+  - `true` - Adorn with an extra visual such as a ring and/or shadow, based on the `activeDisplay` property.
+
+### Interaction
+
+The Avatar is non-interactive.
+
+- **Keyboard** - Not keyboard focusable.
+- **Mouse**
+  - Hover: Show tooltip (TODO: Need to figure out how the tooltip content will be generated based on the name and status).
```

Comment:
We should get some a11y expertise on hover tooltips when the component itself is not focusable at all

Reply:
CCing @smhigley.

Reply:
usually it is more than a tooltip - persona card typically opens on avatar hover. Making that accessible has been challenging in lists and trees but I think it exceeds the scope of Avatar

Reply:
I've logged #16670 to follow up about accessibility